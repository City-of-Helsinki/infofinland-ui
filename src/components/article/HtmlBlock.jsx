import ParseHtml from '../ParseHtml'
import Block from '@/components/layout/Block'
import cls from 'classnames'

const HtmlBlock = ({ field_text }) => (
  <Block className={cls('my-8 ifu-article__bodyblock', field_text?.format)}>
    <ParseHtml html={field_text?.processed} />
  </Block>
)

export default HtmlBlock
