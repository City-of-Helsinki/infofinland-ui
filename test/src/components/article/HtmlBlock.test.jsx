/**
 * @jest-environment jsdom
 */

import React from 'react'
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react'
import HtmlBlock from '@/components/article/HtmlBlock'

const FIELD_CONTENT = {
  format: 'full_html',
  processed: '<p>Paragraph content<a href="http://link.ki/fi">linkki</a></p>',
}

describe('Block', () => {
  it('should render the given HTML inside a Block', () => {
    const { asFragment } = render(
      <HtmlBlock field_text={FIELD_CONTENT} className="tester" />
    )
    // we can only use toBeInTheDocument because it was imported
    // in the jest.setup.js and configured in jest.config.js
    expect(asFragment()).toMatchSnapshot()
  })
})
