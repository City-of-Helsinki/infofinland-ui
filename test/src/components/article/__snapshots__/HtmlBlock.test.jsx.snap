// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block should render the given HTML inside a Block 1`] = `
<DocumentFragment>
  <div
    class="px-6 lg:mx-12 xl:mx-28 2xl:mx-48 3xl:ms-64 3xl:max-w-4xl ifu-article__bodyblock full_html tester"
  >
    <p
      class="mb-8"
    >
      Paragraph content
      <a
        class="ifu-text-link font-bold"
        href="http://link.ki/fi"
      >
        linkki
      </a>
    </p>
  </div>
</DocumentFragment>
`;
